tools/cfn2ts/lib/codegen.ts

```diff
@@ -223,6 +206,52 @@ export default class CodeGenerator {
       this.code.line(`public static readonly REQUIRED_TRANSFORM = ${JSON.stringify(spec.RequiredTransform)};`);
     }
 
+    // true if the resource class has no required properties
+    const onlyOptionalProps = spec.Properties &&
+      !Object.values(spec.Properties).some(p => p.Required || false);
```

Comment:
`??` is a better habit to get into (even though in this case it wouldn't make any difference)

Reply:
Does `??` handle `null`s correctly (in addition to `undefined`)? Because if it doesn't, that's an argument for using `||` still.

Reply:
It does.

Reply:
Alright. Will change.